src/tasks/run_closed_loop.py

```diff
@@ -1,5 +1,45 @@
 """Run all components of the BCI closed loop."""
+import logging
+import os
 import subprocess
+import tempfile
+
+from neural_data_simulator.settings import LogLevel
+from neural_data_simulator.util.runtime import configure_logger
+from neural_data_simulator.util.runtime import initialize_logger
+
+SCRIPT_NAME = "nds-run-closed-loop"
+logger = logging.getLogger(__name__)
+
+
+def _terminate_process(label: str, popen_process: subprocess.Popen, timeout: int = 5):
+    logger.info(f"Terminating {label}")
+    popen_process.terminate()
+    try:
+        popen_process.wait(timeout)
+    except subprocess.TimeoutExpired:
+        logger.warning(f"{label} did not terminate in time. Killing it")
+        popen_process.kill()
+
+
+def _create_pipe_file(pipe_name: str):
+    temp_dir = tempfile.mkdtemp()
+    pipe_path = os.path.join(temp_dir, pipe_name)
+    os.mkfifo(pipe_path)
+    return (temp_dir, pipe_path)
+
+
+def _wait_for_main_task_finish(pipe_path: str):
+    try:
+        with open(pipe_path, "r") as center_out_reach_pipe:
+            for line in center_out_reach_pipe:
+                if "main_task_finished" in line:
+                    logger.info("Main task finished")
+                    break
```

Comment:
```suggestion
            while line := center_out_reach_pipe.readline():
                if "main_task_finished" in line:
                    logger.info("Main task finished")
                    break
```

Reply:
yes, I would also use a `for` when reading a file, but as in this case, we have a **pipe** it seemed to me that a `while` would make it look more like reading from a buffer.
But this is mostly a style suggestion, feel free to leave the `for` if you feel like it

Reply:
gotcha! I ended up including a while True, as we needed to switch to temporary files. Unfortunately, the `while line := ...` approach makes the loop end in the EOF.